src/csharp/Microsoft.Spark.UnitTest/TestUtils/ArrowTestUtils.cs

```diff
@@ -120,82 +67,29 @@ public static ArrowStringDataFrameColumn ToArrowStringDataFrameColumn(StringArra
 
         public static IArrowArray ToArrowArray<T>(T[] array)
         {
-            if (typeof(T) == typeof(bool))
-            {
-                return ToBooleanArray((bool[])(object)array);
-            }
-
-            if (typeof(T) == typeof(sbyte))
-            {
-                return ToPrimitiveArrowArray((sbyte[])(object)array);
-            }
-
-            if (typeof(T) == typeof(byte))
-            {
-                return ToPrimitiveArrowArray((byte[])(object)array);
-            }
-
-            if (typeof(T) == typeof(short))
-            {
-                return ToPrimitiveArrowArray((short[])(object)array);
-            }
-
-            if (typeof(T) == typeof(ushort))
-            {
-                return ToPrimitiveArrowArray((ushort[])(object)array);
-            }
-
-            if (typeof(T) == typeof(int))
-            {
-                return ToPrimitiveArrowArray((int[])(object)array);
-            }
-
-            if (typeof(T) == typeof(uint))
-            {
-                return ToPrimitiveArrowArray((uint[])(object)array);
-            }
-
-            if (typeof(T) == typeof(long))
-            {
-                return ToPrimitiveArrowArray((long[])(object)array);
-            }
-
-            if (typeof(T) == typeof(ulong))
-            {
-                return ToPrimitiveArrowArray((ulong[])(object)array);
-            }
-
-            if (typeof(T) == typeof(float))
-            {
-                return ToPrimitiveArrowArray((float[])(object)array);
-            }
-
-            if (typeof(T) == typeof(double))
-            {
-                return ToPrimitiveArrowArray((double[])(object)array);
-            }
-
-            if (typeof(T) == typeof(DateTime))
-            {
-                return ToPrimitiveArrowArray((DateTime[])(object)array);
-            }
-
-            if (typeof(T) == typeof(TimeSpan))
-            {
-                return ToPrimitiveArrowArray((TimeSpan[])(object)array);
-            }
-
-            if (typeof(T) == typeof(string))
-            {
-                return ToStringArrowArray((string[])(object)array);
-            }
-
-            if (typeof(T) == typeof(byte[]))
-            {
-                return ToBinaryArrowArray((byte[][])(object)array);
-            }
-
-            throw new NotSupportedException($"Unknown type: {typeof(T)}");
+            Type type = typeof(T);
```

Comment:
Let's define `object arrayObject = array;` here and use it below.

Reply:
You didn't implement this suggestion.

Reply:
I am unable to understand your suggestion, can you explain a little bit more? why we have define arrayObject and where it should be used?

Reply:
define `object arrayObject = array;` and replace `(object)array` with `arrayObject`